src/inject/dynamic-theme/css-rules.ts

```diff
@@ -41,23 +42,33 @@ function isCSSVariable(property: string) {
     return property.startsWith('--') && !property.startsWith('--darkreader');
 }
 
+function getParentGroups(rule: CSSRule, stack: string[] = []): string[] {
+    const {parentRule} = rule;
+    if (!parentRule || !(parentRule instanceof CSSMediaRule)) {
+        return stack;
+    }
+    stack.push(parentRule.conditionText);
+    return getParentGroups(parentRule, stack);
+}
+
 export function getCSSVariables(rules: CSSRuleList) {
-    const variables = new Map<string, string>();
+    const variables = new Map<string, Variable>();
     rules && iterateCSSRules(rules, (rule) => {
+        const parentGroups = getParentGroups(rule);
         rule.style && iterateCSSDeclarations(rule.style, (property, value) => {
             if (isCSSVariable(property)) {
-                variables.set(property, value);
+                variables.set(`${rule.selectorText};${property}`, {property, value, parentGroups});
```

Comment:
So, as described above, this should be replaced with
```javascript
if (!variables.has(rule.selectorText)) {
    variables.set(rule.selectorText, new Map());
}
variables.get(rule.selectorText).set(property, {value, parentGroups: []});
```

I'm doubtful about `parentGroups`, but let it be with the first iteration.